vms/avm/vm_test.go

```diff
@@ -1032,296 +462,239 @@ func TestTxCached(t *testing.T) {
 func TestTxNotCached(t *testing.T) {
 	require := require.New(t)
 
-	genesisBytes, _, vm, _ := GenesisVM(t)
-	ctx := vm.ctx
+	env := setup(t, &envConfig{})
 	defer func() {
-		require.NoError(vm.Shutdown(context.Background()))
-		ctx.Lock.Unlock()
+		require.NoError(env.vm.Shutdown(context.Background()))
+		env.vm.ctx.Lock.Unlock()
 	}()
 
-	newTx := NewTx(t, genesisBytes, vm)
+	newTx := newTx(t, env.genesisBytes, env.vm, "AVAX")
 	txBytes := newTx.Bytes()
 
-	_, err := vm.ParseTx(context.Background(), txBytes)
+	_, err := env.vm.ParseTx(context.Background(), txBytes)
 	require.NoError(err)
 
 	registerer := prometheus.NewRegistry()
-	vm.metrics, err = metrics.New("", registerer)
+	env.vm.metrics, err = metrics.New("", registerer)
 	require.NoError(err)
 
 	db := memdb.New()
 	vdb := versiondb.New(db)
-	vm.state, err = states.New(vdb, vm.parser, registerer)
+	env.vm.state, err = states.New(vdb, env.vm.parser, registerer)
 	require.NoError(err)
 
-	vm.uniqueTxs.Flush()
+	env.vm.uniqueTxs.Flush()
 
-	_, err = vm.ParseTx(context.Background(), txBytes)
+	_, err = env.vm.ParseTx(context.Background(), txBytes)
 	require.NoError(err)
 
 	count, err := database.Count(vdb)
 	require.NoError(err)
 	require.NotZero(count)
 }
 
-func TestTxVerifyAfterIssueTx(t *testing.T) {
+func TestTxAcceptAfterParseTx(t *testing.T) {
 	require := require.New(t)
 
-	issuer, vm, ctx, issueTxs := setupIssueTx(t)
+	env := setup(t, &envConfig{
+		notLinearized: true,
+	})
 	defer func() {
-		require.NoError(vm.Shutdown(context.Background()))
-		ctx.Lock.Unlock()
+		require.NoError(env.vm.Shutdown(context.Background()))
+		env.vm.ctx.Lock.Unlock()
 	}()
-	firstTx := issueTxs[1]
-	secondTx := issueTxs[2]
-	parsedSecondTx, err := vm.ParseTx(context.Background(), secondTx.Bytes())
-	require.NoError(err)
-	require.NoError(parsedSecondTx.Verify(context.Background()))
-	_, err = vm.IssueTx(firstTx.Bytes())
-	require.NoError(err)
-	require.NoError(parsedSecondTx.Accept(context.Background()))
-	ctx.Lock.Unlock()
 
-	require.Equal(common.PendingTxs, <-issuer)
-	ctx.Lock.Lock()
-
-	txs := vm.PendingTxs(context.Background())
-	require.Len(txs, 1)
+	key := keys[0]
+	firstTx := &txs.Tx{Unsigned: &txs.BaseTx{
+		BaseTx: avax.BaseTx{
+			NetworkID:    constants.UnitTestID,
+			BlockchainID: chainID,
+			Ins: []*avax.TransferableInput{{
+				UTXOID: avax.UTXOID{
+					TxID:        env.genesisTx.ID(),
+					OutputIndex: 2,
```

Comment:
why is `OutputIndex` 2 not completely apparent from previous test. Looking at Outs I only see [0]

Reply:
This was just copied over from `setupIssueTx`. Any value in [0, 1, 2] would work here. The `OutputIndex` here is referring the the output from the tx it is consuming (`env.genesisTx`).